engine/common/splitter/network/subscription_manager.go

```diff
@@ -0,0 +1,80 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/engine/common/splitter"
+	"github.com/onflow/flow-go/network"
+)
+
+// ChannelSubscriptionManager manages subscriptions of engines running on the node to channels.
+// Each channel should be taken by at most a single engine.
+type SubscriptionManager struct {
```

Comment:
Why the need for a new SubscriptionManager? Why not just use the default one?

Reply:
We need this because the default one is for the default `network` implementation (which uses the "real" network middleware). The main reason I defined a new one here, is because we need the `Channels()` and `GetEngine()` functionality for testing and for the unstaked access node setup later on. For example, the relay engine will need to know the `Channels()` that have been subscribed, so that it can subscribe to all of them on the unstaked network.

Reply:
Not sure why you don't have a middleware. Both the `staked network` and the unstaked network` will need a middleware, it is what actually subscribes using libp2p. 

If you need if purely for testing then you can use a mock. 

`Channels()` is something that the SubcriptionManager current supports.

Reply:
Let me take another look at this. Perhaps what I need to do is implement a new middle ware and use the existing subscription manager.

Reply:
077b265